src/eredis_cluster.erl

```diff
@@ -530,11 +533,11 @@ optimistic_locking_transaction(WatchedKey, GetCommand, UpdateFunction) ->
 %% try again.
 %% @end
 %% =============================================================================
--spec eval(Script::bitstring(), ScriptHash::bitstring(), Keys::[bitstring()],
-           Args::[bitstring()]) -> redis_result().
+-spec eval(Script :: bitstring(), ScriptHash :: bitstring(), Keys :: [bitstring()],
+           Args :: [bitstring()]) -> redis_result().
 eval(Script, ScriptHash, Keys, Args) ->
     KeyNb = length(Keys),
-    EvalShaCommand = ["EVALSHA", ScriptHash, KeyNb] ++ Keys ++ Args,
+    EvalShaCommand = lists:append([["EVALSHA", ScriptHash, KeyNb], Keys, Args]),
```

Comment:
Why append instead of ++? Append is implemented using ++ anyway.

Reply:
It solved the error "538: Function eval/4 has no local return"
Seems that KeyNb is the problem, and it might be better to convert that integer to a string/array using integer_to_list(KeyNb).